homeassistant/components/suez_water/sensor.py

```diff
@@ -28,21 +23,22 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Suez Water sensor from a config entry."""
-    client = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([SuezSensor(client, entry.data[CONF_COUNTER_ID])], True)
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([SuezAggregatedSensor(coordinator, entry.data[CONF_COUNTER_ID])])
 
 
-class SuezSensor(SensorEntity):
+class SuezAggregatedSensor(CoordinatorEntity, SensorEntity):
```

Comment:
```suggestion
class SuezAggregatedSensor(CoordinatorEntity[..], SensorEntity):
```
Fill the generic of the CoordinatorEntity with the type of the coordinator to type the coordinator property